registry/handlers/blobupload.go

```diff
@@ -241,10 +241,11 @@ func (buh *blobUploadHandler) PutBlobUploadComplete(w http.ResponseWriter, r *ht
 
 	desc, err := buh.Upload.Commit(buh, distribution.Descriptor{
 		Digest: dgst,
+		Size:   buh.State.Offset,
```

Comment:
:+1: `buh.State` is only updated at the start of the request and in `blobUploadResponse`, which is not part of `PutBlobUploadComplete`, so this can definitely submit an incorrect size value